src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,208 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.exceptions.responses.UserIdMismatchException;
+import wolox.training.exceptions.responses.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/training/users")
+public class UserController {
+
+    private static final String MSGNOTFOUND = "User not found";
+    private static final String MSGIDMISMATCHED = "User Id mismatched";
+    private static final String MSGSUCCESSFULLYDELETED = "User Successfully Deleted";
+
+    @Autowired
+    private UserRepository userRepository;
+
+    /**
+     * This method is used to get a list with all {@link User}s.
+     *
+     * @return ResponseEntity with a list of {@link User} with all users,
+     *         In case that the list is empty, return ResponseEntity noContent.
+     */
+    @GetMapping
+    public ResponseEntity<List<User>> getAllUsers() {
+        List<User> response = userRepository.findAll();
+        if (response.isEmpty()) {
+            return ResponseEntity.noContent().build();
+        }
+        return ResponseEntity.ok(response);
+    }
+
+    /**
+     * This method is used to get a {@link User} by id.
+     *
+     * @param id: Id of the user to be searched (Long).
+     * @return ResponseEntity with found {@link User} with the id passed.
+     * @exception UserNotFoundException: throw a {@link UserNotFoundException} in case that an {@link User} was not found by the id passed.
+     */
+    @GetMapping("{/id}")
+    public ResponseEntity<User> getUserById(@PathVariable("id") Long id) {
+        Optional<User> response = userRepository.findById(id);
+        if (response.isPresent()) {
+            return ResponseEntity.ok(response.get());
+        } else {
+            throw new UserNotFoundException(MSGNOTFOUND);
+        }
+    }
+
+    /**
+     * This method is used to get an {@link User} by username.
+     *
+     * @param username: Username of the user to be searched (String).
+     * @return ResponseEntity with found {@link User} with the username passed.
+     * @exception UserNotFoundException: throw a {@link UserNotFoundException} in case that an {@link User} was not found by the username passed
+     */
+    @GetMapping("/byUsername")
```

Comment:
Same comment as in a previous pr, you can use the same uri adding RequestParams as needed 😄